src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx

```diff
@@ -119,6 +122,27 @@ export default function DeployStageSetupShell(): JSX.Element {
 
   const { stage: data } = getStageFromPipeline(selectedStageId || '')
 
+  const { data: stageYamlSnippet, loading, refetch } = useGetFailureStrategiesYaml({ lazy: true })
+  React.useEffect(() => {
+    // do the following one if it is a new stage
+    if (!loading && data?.stage && isEmpty(data?.stage?.spec?.execution)) {
+      if (!stageYamlSnippet?.data) {
+        // fetch data on first load of new stage
+        refetch()
+      } else {
+        // update the new stage with the fetched data
+        updateStage(
+          produce(data?.stage as StageElementConfig, draft => {
+            const jsonFromYaml = YAML.parse(defaultTo(stageYamlSnippet?.data, '')) as StageElementConfig
+            if (!draft.failureStrategies) {
+              draft.failureStrategies = jsonFromYaml.failureStrategies
+            }
+          })
+        )
+      }
+    }
+  }, [stageYamlSnippet])
```

Comment:
Codacy has a fix for the issue: [React Hook React.useEffect has missing dependencies: 'data?.stage', 'loading', 'refetch', and 'updateStage'. Either include them or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232284)
```suggestion
  }, [data?.stage, loading, refetch, stageYamlSnippet, updateStage])
```